daft/datatype.py

```diff
@@ -359,6 +366,9 @@ def from_arrow_type(cls, arrow_type: pa.lib.DataType) -> DataType:
             return cls.decimal128(arrow_type.precision, arrow_type.scale)
         elif pa.types.is_date32(arrow_type):
             return cls.date()
+        elif pa.types.is_time64(arrow_type):
+            timeunit = TimeUnit.from_str(pa.type_for_alias(str(arrow_type)).unit)
```

Comment:
CI on pyarrow 6.0.1 was failing because the pyarrow time64 type was represented as `DataType(time64[us])`, which doesnt expose the `unit` attribute as opposed to `Time64Type(time64[us])`. So this is a pretty hacky way of extracting the unit.